hack/gen-swagger-doc/gen-swagger-docs.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Script to generate docs from the latest swagger spec.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo $1
+echo $2
+
+cd /build/
+
+wget $1 -O input.json
+wget $2 -O register.go
+
+./gradle-2.5/bin/gradle gendocs --info
+
+#insert a TOC for top level API objects
+buf="== Top Level API Objects\n\n"
+top_level_models=$(grep IsAnAPIObject ./register.go | sed 's/func (\*\(.*\)) IsAnAPIObject/\1/g' \
+    | tr -d '()' | tr -d '{}' | tr -d ' ')
+for m in $top_level_models
+do
+  if grep -xq "=== v1.$m" ./definitions.adoc
+  then
+    buf+="* <<v1."$m">>\n"
+  fi
+done
+sed -i "1i $buf" ./definitions.adoc
+
+#fix the links in .adoc, replace <<x.y>> with link:definitions.html#_x_y[x.y], and lowercase the _x_y part
+sed -i -e 's|<<\(.*\)\.\(.*\)>>|link:#_\L\1_\2\E[\1.\2]|g' ./definitions.adoc
+sed -i -e 's|<<\(.*\)\.\(.*\)>>|link:definitions.html#_\L\1_\2\E[\1.\2]|g' ./paths.adoc
```

Comment:
Those links are relative links, so it depends on the two file locate in the same folder. It works fine on my local machine.